ChangeLog.md

```diff
@@ -25,6 +25,11 @@ See docs/process.md for more on how version tagging works.
   `-sINCOMING_MODULE_JS_API` instead. (#17955)
 - aio.h was removed from the sysroot.  Emscripten doesn't support any of the
   functions in this header.
+- Clang now has stricter warnings about function pointer casts. The warning helps
+  with CFI errors but also helps wasm builds since wasm traps on such type
+  mismatches in indirect calls. Note that this new clang warning may cause
+  errors on existing code, in which case you may need
+  `-Wno-cast-function-type-strict`. See https://reviews.llvm.org/D134831
```

Comment:
I had to read the LLVM link a couple times and try it out but I think the strict warning is off by default even with `-Wall` and `-Wextra` but is on with `-Wcast-function-type` but it can be disabled (i.e. just get the old less-strict function cast warnings) with `-Wno-cast-function-type-strict`. So maybe we could tweak the wording here to note that  `-Wcast-function-type` is now stricter, and recommend that users try it and fix code that it warns about since it could result in wasm traps;  (and note that the strictness can also be reduced).

Reply:
How about the following:
Clang's function pointer cast warnings (enabled with `-Wcast-function-type`) are now stricter. This warning is intended to help with CFI errors but also helps wasm builds since wasm traps on such type mismatches in indirect calls. We recommend that users enable it to prevent such errors. The older (less strict) behavior can also be enabled with `-Wcast-function-type -Wno-cast-funtion-type-strict` (or `-Wno-error=cast-function-type-strict` if you want the warnings to be visible but not errors)

Reply:
Thanks, rewritten using that.